### PR TITLE
added esConsumes to modules in RecoLocalCalo/EcalRecProducers

### PR DESCRIPTION
**Pull request by JamminJones**
#### PR description:

added esConsumes to 10 modules

#### PR validation:

the code compiles

### CODE REVIEW

**Comment by JamminJones** [comment]
The modules EcalUncalibRecHitWorkerGlobal and EcalUncalibRecHitWorkerWeights were not modified due to complexity of the changes required. Furthermore, the module EcalRecHitWorkerRecover was already modified in the pull request Change EcalTPGScale to use esConsumes.

**Comment by JamminJones** [comment]
The package modified was actually RecoLocalCalo/EcalRecProducers, not RecoLocalCalo/EcalRecHitProducers.

**Comment by slava77** [comment]
> The package modified was actually RecoLocalCalo/EcalRecProducers, not RecoLocalCalo/EcalRecHitProducers.

there is an "Edit" button near the PR title in the web interface.
If the title is wrong, it can be edited.

**Comment by Dr15Jones** [comment]
This change was made under my supervision.

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
The failure in test 4.53 is
```
An exception of category 'ESGetTokenWrongTransition' occurred while
   [0] Processing  Event run: 194533 lumi: 329 event: 461582793 stream: 0
   [1] Running path 'dqmoffline_step'
   [2] Prefetching for module DQMMessageLogger/'DQMMessageLogger'
   [3] Prefetching for module LogErrorHarvester/'logErrorHarvester'
   [4] Calling method for module EcalTPSkimmer/'ecalTPSkim'
Exception Message:
The transition ID stored in the ESGetToken does not match the
transition where the token is being used. The associated record
type is: EcalChannelStatusRcd
For producers, filters and analyzers this transition ID is
set as a template parameter to the call to the esConsumes
function that creates the token. Event is the default transition.
Other possibilities are BeginRun, EndRun, BeginLuminosityBlock,
or EndLuminosityBlock. You may need multiple tokens if you want to
get the same data in multiple transitions. The transition ID has a
different meaning in ESProducers. For ESProducers, the transition
ID identifies the function that produces the EventSetup data (often
there is one function named produce but there can be multiple
functions with different names). For ESProducers, the ESGetToken
must be used in the function associated with the ESConsumesCollector
returned by the setWhatProduced function.
```

File: RecoLocalCalo/EcalRecProducers/plugins/EcalTPSkimmer.cc [hunk 1]
```diff
@@ -31,7 +26,10 @@ EcalTPSkimmer::EcalTPSkimmer(const edm::ParameterSet& ps) {
 
   tpOutputCollection_ = ps.getParameter<std::string>("tpOutputCollection");
   tpInputToken_ = consumes<EcalTrigPrimDigiCollection>(ps.getParameter<edm::InputTag>("tpInputCollection"));
-
+  ttMapToken_ = esConsumes<EcalTrigTowerConstituentsMap, IdealGeometryRecord>();
+  if (skipModule_) {
```
Comment (Dr15Jones) [inline RecoLocalCalo/EcalRecProducers/plugins/EcalTPSkimmer.cc:30]:
```suggestion
  if (not skipModule_) {
```

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitWorkerSimple.cc [hunk 2]
```diff
@@ -26,6 +19,16 @@ EcalRecHitWorkerSimple::EcalRecHitWorkerSimple(const edm::ParameterSet& ps, edm:
...
+  if (!skipTimeCalib_) {
+    itimeToken_ = c.esConsumes<EcalTimeCalibConstants, EcalTimeCalibConstantsRcd>();
+    offtimeToken_ = c.esConsumes<EcalTimeOffsetConstant, EcalTimeOffsetConstantRcd>();
+  }
+  agcToken_ = c.esConsumes<EcalADCToGeVConstant, EcalADCToGeVConstantRcd>();
+  chStatusToken_ = c.esConsumes<EcalChannelStatus, EcalChannelStatusRcd>();
+  if (laserCorrection_)
+    laserToken_ = c.esConsumes<EcalLaserDbService, EcalLaserDbRecord>();
+
   // Traslate string representation of flagsMapDBReco into enum values
```
Comment (slava77) [inline RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitWorkerSimple.cc:32]: shouldn't ```EcalRecHitWorkerSimple::set``` be updated as well to use getHandle?

**Comment by slava77** [comment]
is there going to be another installment to cover the remaining 3 modules in this package?
- RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerGlobal.cc
- RecoLocalCalo/EcalRecProducers/plugins/EcalDetailedTimeRecHitProducer.cc
- RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerWeights.cc

**Comment by Dr15Jones** [comment]
@slava77 the developer hasn't been feeling well for the past couple of days so there is likely to be a delay in making modifications.

**Comment by Dr15Jones** [comment]
@slava77 
> is there going to be another installment to cover the remaining 3 modules in this package?
> 
> * RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerGlobal.cc
> * RecoLocalCalo/EcalRecProducers/plugins/EcalDetailedTimeRecHitProducer.cc
> * RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerWeights.cc

EcalDetailedTimeRecHitProducer was an oversight and has been added to this pull request. The other two are much more involved so will not be handled by the developer. There is no plan for Core to make changes to those modules. Once the developer is done for the Summer, we intend to create an issue containing all remaining modules we know of which are used in production and need to be updated. This will hopefully allow the L2s responsible for those modules to find individuals to make the needed changes.

**Comment by Dr15Jones** [comment]
please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/30843	5752061 
- code changes are in line with the PR description and the follow up review
    - the remaining part of the package will need migration in the future as proposed in https://github.com/cms-sw/cmssw/pull/30843#issuecomment-663600209
- jenkins tests pass and comparisons with the baseline show no (relevant) differences

**Comment by silviodonato** [comment]
+1

**Pull request merged**